### PR TITLE
StatPanels: Refactoring DisplayValueOptions and renaming & adding new panel options to react panels 

### PR DESCRIPTION
**Pull request by torkelo**
So after moving fieldConfig out of react options.fieldOptions we still have the `values`, `calcs`,  `limit` options on the options.fieldOptions the naming makes little sense as there are not really fieldOptions but more like single stat "reduce" options.

* Renaming (this was hard feedback / alternatives are welcome)
* Trying to use the new panel option registry to set these common value option properties (needs more work for the calc/stat picker)  
* the panel options registry now supports deep key paths like `valueOptions.values` in order to still have these options defined in it's own object while keeping the option items in a flat list 

Adds panel options (using registry) for stat panel options as well.

### CODE REVIEW

File: packages/grafana-data/src/field/fieldDisplay.ts [hunk 1]
```diff
@@ -20,15 +19,16 @@ import { reduceField, ReducerID } from '../transformations/fieldReducer';
...
-  limit?: number; // if showing all values limit
-  calcs: string[]; // when !values, pick one value for the whole field
-  override?: any;
-  defaults?: {
-    mappings: ValueMapping[];
-  };
+/**
+ * Options for how to turn DataFrames into an array of display values
+ */
+export interface DataToSingleValueOptions {
```
Comment (torkelo) [inline packages/grafana-data/src/field/fieldDisplay.ts]: naming this type is hard

File: public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx [hunk 2]
```diff
@@ -228,7 +228,7 @@ export class PanelEditorUnconnected extends PureComponent<Props> {
     const { plugin, dashboard, data, panel } = this.props;
 
     if (!plugin) {
-      return null;
+      <div />;
```
Comment (ryantxu) [inline public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx]: missing return?

**Comment by torkelo** [comment]
This should be good now, now all react panels has all the options they have in old edit mode but using new panel option structure

**Comment by torkelo** [comment]
Todo: rename DataToSingleValueOptions to ReduceDataOptions and panel options property from valueOptions -> reduceOptions

**Review by ryantxu — APPROVED** [review]
LGTM

**Pull request merged**